### PR TITLE
CI & linting

### PR DESCRIPTION
**Pull request by PeterJCLaw**
Adds CI linting and fixes a bunch of issues. Note that this also disables some lints which don't appear entirely useful.

I'm open to the list of disabled lints changing; though I think there's value in getting this merged sooner and then changing them later, rather than debating that here.

### CODE REVIEW

File: robotd/devices.py [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 import random
 import struct
 import subprocess
 import time
+from pathlib import Path
 
-from sb_vision import Camera as VisionCamera, Vision, Token
 import serial
+from sb_vision import Camera as VisionCamera
+from sb_vision import Token, Vision
```
Comment (RealOrangeOne) [inline robotd/devices.py]: Should these lines be merged?

`from sb_vision import Camera as VisionCamera, Token, Vision`

File: robotd/devices.py [hunk 2]
```diff
@@ -349,10 +359,14 @@ def _command(self, *args):
...
                 if line.startswith(b'@'):
                     returned_command_id_str, line = line[1:].split(b' ', 1)
-                    returned_command_id = int(returned_command_id_str.decode('utf-8')) & 0xffff
+                    returned_command_id = int(
+                        returned_command_id_str.decode('utf-8'),
+                    ) & 0xffff
 
                     if returned_command_id != command_id:
-                        print('Got response for different command, ignoring...')
+                        print(
```
Comment (RealOrangeOne) [inline robotd/devices.py]: Oh, What fresh hell is this?! 😱

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[flake8]
+exclude =
+    .git,
```
Comment (RealOrangeOne) [inline setup.cfg:3]: Are all these ignores really necessary? If we're specifying the module/directory, surely it should only check that directory, and know not to lint the `__pycache__` dir?

Comment (PeterJCLaw) [reply setup.cfg:3]: flake8's docs include the suggestion of including the `__pycache__` and `.git` directories.
I mainly included them as I'm not sure what arguments editors will call flake8 with.

File: robotd/devices.py (hunk 2 above)
Comment (PeterJCLaw) [reply robotd/devices.py]: How else would you wrap this? (This is the style we use at work; I agree it looks a bit odd here, but it works _really_ well for longer or more complicated examples)

File: robotd/devices.py (hunk 1 above)
Comment (PeterJCLaw) [reply robotd/devices.py]: Maybe, though this is what `isort` generated.

File: robotd/devices.py (hunk 2 above)
Comment (RealOrangeOne) [reply robotd/devices.py]: My main issue was the trailing comma on the arguments. 

I think my eyes are used to seeing Python aligned to 120 length over 80, which makes this break seem unnecessary, but that's just my training.

Comment (PeterJCLaw) [reply robotd/devices.py]: Ah, I see. This is from `flake8-commas` which is handy as it ensures that when adding more things you don't get the diff noise from adding commas on the preceding lines.

I'm open to making the boundary a little longer (79 is the default), though I'm not sure I'd go as far as 120 characters. Maybe 90?

Comment (RealOrangeOne) [reply robotd/devices.py]: I _really_ don't like trailing commas like that. They make sense on multi-element, multi-line cases, but in this case, it just looks wrong. Although, i'm fairly happy to be outvoted, assuming the linter enforces whatever standard we decide on.

79 for me is always a little too short. I think 90 works for me! On a slightly unrelated note, would be good to enforce the same linter rules on all repos

Comment (PeterJCLaw) [reply robotd/devices.py]: Yeah, if we didn't have a hard limit on line lengths I'd definitely leave this unwrapped.

3e16f97

File: robotd/devices.py (hunk 1 above)
Comment (RealOrangeOne) [reply robotd/devices.py]: hmm, looks like it's doing this split all over the place. Assuming it's standardised, i'm fine with it

File: robotd/game_specific.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+"""Game specific values."""
```
Comment (RealOrangeOne) [inline robotd/game_specific.py]: I think this file has been committed accidentally. This file was moved to `sb-vision`

Comment (RealOrangeOne) [inline robotd/game_specific.py]: I think this file has been committed accidentally. This file was moved to `sb-vision`

**Review by RealOrangeOne — CHANGES_REQUESTED** [review]
Looks like you've accidentally committed an old file!

File: robotd/game_specific.py (hunk 4 above)
Comment (Adimote) [reply robotd/game_specific.py]: +1

File: robotd/devices.py (hunk 1 above)
Comment (PeterJCLaw) [reply robotd/devices.py]: Yeah, I'm not sure I like it either. As noted in the PR I'd like to merge this soon and then sort out our preferences later. Especially in this case changing later is easy as you can just run `isort` on all the files and it'll convert over.

File: robotd/game_specific.py (hunk 4 above)
Comment (PeterJCLaw) [reply robotd/game_specific.py]: Huh, this file was there when I started the branch, though it did get removed by a merge. Annoying that it's come back.

Comment (PeterJCLaw) [reply robotd/game_specific.py]: 01440da

**Review by RealOrangeOne — APPROVED** [review]

**Pull request merged**